CondCore/Utilities/plugins/BasicP_PayloadInspector.cc

```diff
@@ -91,6 +96,88 @@ namespace {
     }
   };
 
+  class BasicPayload_data6 : public cond::payloadInspector::PlotImage<cond::BasicPayload> {
+  public:
+    BasicPayload_data6() : cond::payloadInspector::PlotImage<cond::BasicPayload>( "Example delivery picture" ){
+      setSingleIov( true );
+    }
+
+    bool fill( const std::vector<std::tuple<cond::Time_t,cond::Hash> >& iovs ){
+      auto iov = iovs.front();
+      std::shared_ptr<cond::BasicPayload> payload = fetchPayload( std::get<1>(iov) );
+  
+      double xmax(100.),ymax(100.);
+
+      TH2D *h2D = new TH2D("h2D","Example",100,0.,xmax,100,0.,ymax);
```

Comment:
Hi David, since I'm not a root expert, this code has been provided 'externally'.  While integrating this in the package, I was also wondering if some clean up was needed, but i all the examples I found searching around this wasn't done! 
However, I admit I did not try to improve the 'imported' code along this line (or to make unique_ptr, which has the same effect ), so It maybe well possible that you are totally right.
Please et me know if you know more about that, in any case I'll do some test...